portal-bridge/src/bridge/history.rs

```diff
@@ -193,9 +201,20 @@ impl HistoryBridge {
                 None
             };
             info!("fetching headers in range: {gossip_range:?}");
-            self.serve(gossip_range, epoch_acc)
-                .await
-                .expect("Error serving headers in backfill mode.");
+            for height in gossip_range.clone() {
+                let cloned_self = self.clone();
```

Comment:
Is this a good idea to simply clone the entire bridge in order to enable spawning threads for each new block to be gossiped?